src/rpc/rpc.cc

```diff
@@ -977,6 +978,9 @@ namespace rpc {
               } else {
                   abort_all_streams();
               }
+          }).finally([this] {
+            _compressor_needs_progress.broken();
+            return std::move(_compressor_bumper_exit);
```

Comment:
Don't you need to add the same for the server loop?

Reply:
Right, I do. Good catch.

Reply:
Fixed.